radix-engine-common/src/address/hrpset.rs

```diff
@@ -5,20 +5,26 @@ use sbor::rust::prelude::*;
 /// Represents an HRP set (typically corresponds to a network).
 #[derive(Debug, Clone)]
 pub struct HrpSet {
-    package: String,
-    resource: String,
-    component: String,
-    account: String,
-    identity: String,
-    consensus_manager: String,
-    validator: String,
-    access_controller: String,
-    pool: String,
-    transaction_tracker: String,
-    internal_vault: String,
-    internal_account: String,
-    internal_component: String,
-    internal_key_value_store: String,
+    /* Entities */
+    pub package: String,
+    pub resource: String,
+    pub component: String,
+    pub account: String,
+    pub identity: String,
+    pub consensus_manager: String,
+    pub validator: String,
+    pub access_controller: String,
+    pub pool: String,
+    pub transaction_tracker: String,
+    pub internal_vault: String,
+    pub internal_account: String,
+    pub internal_component: String,
+    pub internal_key_value_store: String,
+
+    /* Transaction Parts */
+    pub transaction_intent: String,
+    pub signed_transaction_intent: String,
+    pub notarized_transaction: String,
```

Comment:
So we also have quite a few different transaction hashes in the node (ledger, round update, genesis).

So I wonder if (to be extensible) we can instead put these (network-independent) prefixes on the `HashHasHrp`? (maybe call it `TransactionHash`)? And then combine them with the network suffix inside the TransactionBech32Encoder/Decoder?

Reply:
Why not extend the HRP set to include the other transaction types such as ledger transactions, round update, genesis, etc... and then implement the `HashHasHrp` trait in the node?